internal/core/users/users.repository.go

```diff
@@ -54,6 +62,32 @@ func (repo *repositoryImpl) UpdateUser(user *models.UpdatingUserPersonalInfo, us
 	return repo.db.Where("user_id = ?", userId).Updates(user).Error
 }
 
+func (repo *repositoryImpl) UpdateUserFinancialInformationById(userFinancialInformation *models.UserFinancialInformations, userId string) error {
+	err := repo.db.First(&models.UserFinancialInformations{}, "user_id = ?", userId).Error
+	if err != nil {
+		return err
+	}
+
+	return repo.db.Transaction(func(tx *gorm.DB) error {
+		var countCreditCards int64
+		if err := tx.Model(&models.CreditCards{}).Where("user_id = ?", userId).Count(&countCreditCards).Error; err != nil {
+			return err
+		} else if countCreditCards > 0 {
+			if err := tx.Where("user_id = ?", userId).Delete(&models.CreditCards{}).Error; err != nil {
+				return err
+			}
+		}
+
+		for _, creditCard := range userFinancialInformation.CreditCards {
+			if err := tx.Model(&models.CreditCards{}).Create(&creditCard).Error; err != nil {
+				return err
+			}
+		}
```

Comment:
There's an alternative method using `db.CreateInBatches` check it out here [https://gorm.io/docs/create.html#Batch-Insert](https://gorm.io/docs/create.html#Batch-Insert)
However, this is `not a must-change stuff`. I'm just pointing out that gorm supports batches creation operation. The querys are already wrapped in a transaction. So, that's already fantastic.

Reply:
I have changed to `db.Create(&[]models.CreditCards{})`, and it works though.